fn_alienvault_otx/fn_alienvault_otx/components/fn_alienvault_otx_threat_lookup.py

```diff
@@ -0,0 +1,103 @@
+
+            alienvault_section = kwargs.get("alienvault_section")  # text
+            alienvault_section = alienvault_section.strip() if alienvault_section else None
+
+            log = logging.getLogger(__name__)
+            log.info("alienvault_search_value: %s", alienvault_search_value)
+            log.info("alienvault_search_type: %s", alienvault_search_type)
+            log.info("alienvault_section: %s", alienvault_section)
+
+            yield StatusMessage("starting...")
+            alien_vault_get_url = ApiCallController.create_alienvault_indicators_url(self.options.get('av_base_url'),
+                                                                                     alienvault_search_value,
+                                                                                     alienvault_search_type,
+                                                                                     alienvault_section)
+            # Getting Alien Vault API Key From app.config
+            AV_API_KEY = self.options.get('av_api_key')
+
+            # Getting Proxy Settings from app.config
+            PROXY = self.options.get('proxy')
+
+            # Creating API Call Header
+            CALL_HEADER = ApiCallController.create_header(api_key=AV_API_KEY)
+
+            # proxy data to be sent through the api call
+            AV_PROXY = ApiCallController.format_proxy_data(proxy_data=PROXY)
+
+            # Api Controller Class Instance
+            ApiCallController_instance = ApiCallController()
+
+            # initialising the Session
+            _request_session = ApiCallController_instance.session()
+
+            # Connection to Alien Vault OTX to make a GET request call for given URL.
+            _api_response = _request_session.get(alien_vault_get_url, headers=CALL_HEADER, proxies=AV_PROXY)
+
+            _api_response.raise_for_status()
+
+            # api_response_json = ApiCallController_instance.response_handle_errors(response=_api_response).json()
+            # Just to check received response is json object
+            _api_response_json = _api_response.json()
+
+            # Converts a String Timestamp to an int in milliseconds since 1970 epoch
+            _api_response_text = _api_response.text
+            _api_response_text = ApiCallController.convert_date_string_to_ms_epoch(response_data=_api_response_text)
+
+            # converting string response to json object after changing the time format
+            try:
+                _api_response_json = json.loads(_api_response_text)
+            except Exception as e:
+                log.info("Error occurred while converting time stamp to epoch, time stamp is not converted")
+
+            results = {
+                "content": _api_response_json
+            }
+            log.info("API CALL URL : {}".format(alien_vault_get_url))
+
+            # Closing Connection
+            requests.session().close()
```

Comment:
we haven't been doing this in other code. But doesn't mean it's not useful. Is the session closed when the sw completes?

One consideration we may have to be careful about is this module is shared with other modules in memory. Is requests.session().close() thread safe?